project libs2e/
diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 4a1d8ce..e4457b3 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -36,7 +36,8 @@ endif()
 
 set(LIBS ${LIBS} ${LIBCOROUTINE_LIBRARY_DIR}/libcoroutine.a pthread glib-2.0 bsd)
 
-
+# hope this works
+set(LIBS ${LIBS} crypto ssl)
 
 target_link_libraries(tests2e PUBLIC ${LIBCOROUTINE_LIBRARY_DIR}/libcoroutine.a pthread glib-2.0)
 target_link_libraries(s2e ${LIBS})
@@ -57,4 +58,5 @@ set(COMMON_FLAGS "${COMMON_FLAGS} -Wall -fPIC -fno-strict-aliasing -fexceptions"
 
 set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${WERROR_FLAGS} ${COMMON_FLAGS} -std=c++11")
 set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} ${WERROR_FLAGS} ${COMMON_FLAGS}")
-set(CMAKE_SHARED_LINKER_FLAGS "-Wl,--version-script=${CMAKE_CURRENT_SOURCE_DIR}/mapfile")
+# set(CMAKE_SHARED_LINKER_FLAGS "-Wl,--version-script=${CMAKE_CURRENT_SOURCE_DIR}/mapfile")
+set(CMAKE_SHARED_LINKER_FLAGS "-lssl -lcrypto -Wl,--version-script=${CMAKE_CURRENT_SOURCE_DIR}/mapfile")

project libs2eplugins/
diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 37b82ad..ea56585 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -69,6 +69,8 @@ add_library(
     s2e/Plugins/Searchers/SeedScheduler.cpp
     s2e/Plugins/Searchers/PathSearcher.cpp
 
+    s2e/Plugins/Searchers/ConcolicExploreSearcher.cpp
+
     # Function models
     s2e/Plugins/Models/BaseFunctionModels.cpp
     s2e/Plugins/Models/FunctionModels.cpp
@@ -117,8 +119,9 @@ add_library(
 )
 
 
-target_include_directories (s2eplugins PUBLIC ${CMAKE_CURRENT_SOURCE_DIR})
+target_link_libraries(s2eplugins ctypto)
 
+target_include_directories (s2eplugins PUBLIC ${CMAKE_CURRENT_SOURCE_DIR})
 
 set(WERROR_FLAGS "-Werror -Wno-zero-length-array -Wno-c99-extensions          \
                   -Wno-gnu-anonymous-struct -Wno-nested-anon-types            \
@@ -135,3 +138,8 @@ set(COMMON_FLAGS "${COMMON_FLAGS} -Wall -fPIC -fno-strict-aliasing -fexceptions
 
 set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${WERROR_FLAGS} ${COMMON_FLAGS}")
 set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} ${WERROR_FLAGS} ${COMMON_FLAGS}")
+
+# set(CMAKE_SHARED_LINKER_FLAGS "crypto")
+# set(CMAKE_MODULE_LINKER_FLAGS "-lcrypto")
+# set(CMAKE_STATIC_LINKER_FLAGS "-lcrypto")
+# set(CMAKE_EXEC_LINKER_FLAGS "-lcrypto")
